apps/ocsp.c

```diff
@@ -967,7 +981,22 @@ static void make_ocsp_response(OCSP_RESPONSE **resp, OCSP_REQUEST *req,
 
     OCSP_copy_nonce(bs, req);
 
-    OCSP_basic_sign(bs, rcert, rkey, rmd, rother, flags);
+    mctx = EVP_MD_CTX_new();
+    if ( mctx == NULL || !EVP_DigestSignInit(mctx, &pkctx, rmd, NULL, rkey)) {
+        *resp = OCSP_response_create(OCSP_RESPONSE_STATUS_INTERNALERROR, NULL);
+        goto end;
+    }
+    for (i = 0; i < sk_OPENSSL_STRING_num(sigopts); i++) {
+        char *sigopt = sk_OPENSSL_STRING_value(sigopts, i);
+        if (pkey_ctrl_string(pkctx, sigopt) <= 0) {
+            BIO_printf(err, "parameter error \"%s\"\n", sigopt);
+            ERR_print_errors(bio_err);
+            *resp = OCSP_response_create(OCSP_RESPONSE_STATUS_MALFORMEDREQUEST,
+                                         NULL);
+            goto end;
```

Comment:
Perhaps I chose the wrong error. I haven't used `ocsp` as an OCSP responder, just to create OCSP responses from the command line:
```
openssl ocsp -index CA/index.txt -rsigner signer.pem -rkey signer.key -CA CA/CA.pem -issuer CA/CA.pem -no_nonce -rmd sha256 -rsigopt rsa_padding_mode:pss -rsigopt rsa_pss_saltlen:64 -nmin 120 -respout resp.der -no_cert_verify -serial 1
```
The way that I caused `pkey_ctrl_string` to fail was by providing an invalid parameter to `-rsigopt` (e.g., `-rsigopt rsa_made:pss`), which seemed like a malformed request, since the `-rsigopt` was part of the request I was making. It's not exactly consistent with the [RFC 6960](https://www.rfc-editor.org/rfc/rfc6960.txt) definition of malformedRequest, but malformedRequest seemed to be a better fit than any of the others.

I guess if the response were sent to a remote entity that didn't have any control over the command-line parameters, internalError would seem more appropriate to that remote entity.